Zend/zend_weakrefs.c

```diff
@@ -0,0 +1,217 @@
+/*
+   +----------------------------------------------------------------------+
+   | PHP Version 7                                                        |
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 2.00 of the Zend license,     |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.zend.com/license/2_00.txt.                                |
+   | If you did not receive a copy of the Zend license and are unable to  |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@zend.com so we can mail you a copy immediately.              |
+   +----------------------------------------------------------------------+
+   | Authors: krakjoe@php.net                                             |
+   +----------------------------------------------------------------------+
+*/
+
+#include "zend.h"
+#include "zend_interfaces.h"
+#include "zend_objects_API.h"
+#include "zend_weakrefs.h"
+
+typedef struct _zend_weakref {
+	zval referent;
```

Comment:
Could be just `zend_object*`?

Reply:
I don't understand how that would work ?

Reply:
The referent is always either an object or null. We can just store the object pointer or NULL instead. Use ZVAL_OBJ(return_value, wr->referent) when returning.